GitUI/CommandsDialogs/RepoHosting/CreatePullRequestForm.cs

```diff
@@ -132,31 +132,37 @@ private void PopulateBranchesComboAndEnableCreateButton(IHostedRemote remote, Co
                     {
                         await TaskScheduler.Default;
 
-                        IHostedRepository hostedRepository = remote.GetHostedRepository();
-                        var branches = hostedRepository.GetBranches();
+                        try
+                        {
+                            IHostedRepository hostedRepository = remote.GetHostedRepository();
+                            var branches = hostedRepository.GetBranches();
 
-                        await this.SwitchToMainThreadAsync();
+                            await this.SwitchToMainThreadAsync();
 
-                        comboBox.Items.Clear();
+                            comboBox.Items.Clear();
 
-                        var selectItem = 0;
-                        var defaultBranch = hostedRepository.GetDefaultBranch();
-                        for (var i = 0; i < branches.Count; i++)
-                        {
-                            if (branches[i].Name == defaultBranch)
+                            var selectItem = 0;
+                            var defaultBranch = hostedRepository.GetDefaultBranch();
+                            for (var i = 0; i < branches.Count; i++)
                             {
-                                selectItem = i;
+                                if (branches[i].Name == defaultBranch)
+                                {
+                                    selectItem = i;
+                                }
+
+                                comboBox.Items.Add(branches[i].Name);
                             }
 
-                            comboBox.Items.Add(branches[i].Name);
-                        }
+                            if (branches.Count > 0)
```

Comment:
`branches.Any()`

Reply:
Why? This will be slower.

Reply:
It is more readable, more flexible, and uses a named method to denote an operation.
But I agree, this is subjective.

Reply:
Maybe extension method `Any()` for list will be better. 😄

Reply:
Having worked on the .NET SDK for the past few years I've learnt to avoid Linq.

Reply:
To be clear, my original objection wasn't really about performance. I guess it's what you're used to. I'm perfectly used to seeing a count check. My brain understands immediately, and also doesn't worry about allocations or overhead. If I see `Any` I worry about those things slightly, and worry about multiple enumeration and so forth more. But it's a matter of style.

I find these kinds of style things in PR reviews can provide a micro-barrier to contribution. In general I think projects at this kind of scale should use analyzers to enforce these kinds of preferences to save everyone's time and reduce friction. Of course that sometimes means writing an analyzer :)